fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/QueryPlan.java

```diff
@@ -156,9 +157,10 @@ default boolean isStrictlySorted() {
 
     /**
      * Return a copy of this plan that has the {@link #isStrictlySorted} property.
+     * @param memoizer a memoizer that is used to memoize/re-reference new expressions references
      * @return a copy of this plan
      */
-    default QueryPlan<T> strictlySorted() {
+    default QueryPlan<T> strictlySorted(@Nonnull final Memoizer memoizer) {
```

Comment:
This feels weird™, like it's mixing concerns. If memoization is not the normal concern of plan construction, then you'd expect the caller to have to memoize the sorted result.

If there were a memoized monad (kind of like what we discussed with separate classes), you'd want to bind strict-sorted to the raw plan.

With the current approach, maybe this, like the `Quantifier.physical` can be wrapped in some convenience, either just for this of generally for a `Function<T, T>`.